### PR TITLE
chore: align fetch to spec

### LINKED ISSUE #10002: fetch with Request init with body of FormData is broken
`FormData` set as the `body: BodyInit` of the `RequestInit` passed to the `Request` constructor should behave the same way as the `body: BodyInit` of the `RequestInit` as the second arg to `fetch(input, init)`

Currently it doesn't, sending a request body with the text `[object FormData]`

```
const formData = new FormData();
formData.append('foo', 'bar');
const res = await fetch(new Request('http://example.com', { method: 'POST', body: formData }));
// request.text() on the server side will be "[object FormData]", request.formData() will be empty
```

deno 1.8.2 (release, x86_64-apple-darwin)
v8 9.0.257.3
typescript 4.2.2

### PR DESCRIPTION
**Pull request by lucacasonato**
This commit aligns the `fetch` API and the `Request` / `Response`
classes belonging to it to the spec. This commit enables all the
relevant `fetch` WPT tests. Spec compliance is now at around 90%.

Performance is essentially identical now (within 1% of 1.9).

Closes #10002

### CODE REVIEW

File: op_crates/fetch/21_formdata.js [hunk 1]
```diff
@@ -442,6 +442,11 @@
      * @returns {FormData}
      */
     parse() {
+      // Body must be at least 2 boundaries + \r\n + -- on the last boundary.
+      if (this.body.length < (this.boundary.length * 2) + 4) {
+        throw new TypeError("FormData to short to be valid.");
```
Comment (aapoalas) [inline op_crates/fetch/21_formdata.js:447]: `too short`

File: op_crates/fetch/20_headers.js [hunk 2]
```diff
@@ -118,10 +122,54 @@
...
+    if (initialValue === null) return null;
+    const input = initialValue;
+    let position = 0;
+    const values = [];
+    let value = "";
+    while (position < initialValue.length) {
+      const res = collectSequenceOfCodepoints(
+        initialValue,
+        position,
+        (c) => c !== "\u0022" && c !== "\u002C",
```
Comment (ry) [inline op_crates/fetch/20_headers.js:141]: Tab or space? Add a comment?

File: op_crates/fetch/22_http_client.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+((window) => {
+  const core = window.Deno.core;
+
+  /**
+   * @param {Deno.CreateHttpClientOptions} options
+   * @returns {HttpClient}
+   */
+  function createHttpClient(options) {
+    return new HttpClient(core.opSync("op_create_http_client", options));
+  }
```
Comment (ry) [inline op_crates/fetch/22_http_client.js:24]: Aside: we should get rid of `Deno.createHttpClient`, `new Deno.HttpClient` should be sufficient.

**Review by ry — APPROVED** [review]
LGTM - great work - a leap towards better web compatibility.

File: op_crates/fetch/22_body.js [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    switch (type) {
+      case "ArrayBuffer":
+        return bytes.buffer;
+      case "Blob":
+        return new Blob([bytes], {
+          type: mimeType !== null ? mimesniff.serializeMimeType(mimeType) : "",
+        });
+      case "FormData": {
+        if (mimeType !== null) {
+          if (mimeType !== null) {
```
Comment (CrimsonCodes0) [inline op_crates/fetch/22_body.js:225]: Really late, but the nested `if` statement body is unreachable, you have a duplicate condition.

File: op_crates/fetch/23_request.js [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+      // 29.
+      this[_headers] = headersFromHeaderList(request.headerList, "request");
+
+      // 31.
+      if (Object.keys(init).length > 0) {
+        let headers = headerListFromHeaders(this[_headers]);
+        if (init.headers !== undefined) {
+          headers = init.headers;
+        }
+        headerListFromHeaders(this[_headers]).slice(
```
Comment (CrimsonCodes0) [inline op_crates/fetch/23_request.js:254]: I can't tell which .slice method this is, but does it mutate the source? If not, this is dead code.

**Pull request merged**